prov/gni/src/gnix_vc.c

```diff
@@ -1104,10 +1122,11 @@ static int __gnix_vc_hndl_conn_req(struct gnix_cm_nic *cm_nic,
 	 * will be filled in.
 	 */
 	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
+	fastlock_acquire(&ep->vc_lock);
```

Comment:
The lock needs to be there since __gnix_vc_lookup_vc reads from the vc vector or hashtable.  I’m going to change and test __gnix_vc_lookup_vc so that locks the vector or hashtable internally.  Yes, I will add comments.
> On Apr 11, 2016, at 5:28 PM, Sung-Eun Choi <notifications@github.com> wrote:
> 
> In prov/gni/src/gnix_vc.c <https://github.com/ofi-cray/libfabric-cray/pull/774#discussion_r59299207>:
> 
> > @@ -1104,10 +1122,11 @@ static int __gnix_vc_hndl_conn_req(struct gnix_cm_nic *cm_nic,
> >  	 * will be filled in.
> >  	 */
> >  	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
> > +	fastlock_acquire(&ep->vc_lock);
> Why does this lock need to be here now? Can you put in comments to document for the future?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/ofi-cray/libfabric-cray/pull/774/files/f7f691fdc3534a4839aacfe36a46f37799838774..3f38e6d2180fb715f3a2077087f95fbff8fe2078#r59299207>